### PR TITLE
Fix almost all PHPStan level 5 problems

### PR DESCRIPTION
**Pull request by szepeviktor**
@tillkruss This is the max I can do in a short time.

Please [send the usual $1 000 000](https://szepe.net/wp-donate/). Thanks!

Hire me!! :)

### CODE REVIEW

**Comment by szepeviktor** [comment]
These need ~100 hours to track down.

```
 ------ ----------------------------------------------------
  Line   benchmarks/MgetBench.php
 ------ ----------------------------------------------------
  58     Call to an undefined method Credis_Client::mget().
 ------ ----------------------------------------------------

 ------ ------------------------------------------------------
  Line   src/Laravel/RelayConnection.php
 ------ ------------------------------------------------------
  22     Call to an undefined method Redis::dispatchEvents().
  38     Call to an undefined method Redis::listen().
  49     Call to an undefined method Redis::onFlushed().
  61     Call to an undefined method Redis::onInvalidated().
  71     Call to an undefined method Redis::stats().
  81     Call to an undefined method Redis::license().
  91     Call to an undefined method Redis::socketId().
  103    Call to an undefined method Redis::socketKey().
 ------ ------------------------------------------------------
```

File: tests/Relay.stub [hunk 1]
```diff
@@ -0,0 +1,1796 @@
...
+
+class Relay {
+    public const OPT_COMPRESSION = 1;
+    public const OPT_COMPRESSION_LEVEL = 1;
+    public const OPT_PHPREDIS_COMPATIBILITY = 1;
+    public const OPT_PREFIX = 1;
+    public const OPT_READ_TIMEOUT = 1;
+    public const OPT_SCAN = 1;
+    public const OPT_SERIALIZER = 1;
+    public const SERIALIZER_PHP = 1;
```
Comment (szepeviktor) [inline tests/Relay.stub:18]: It would be nice to fix these constants.

Comment (tillkruss) [reply tests/Relay.stub:18]: Fix how?

File: composer.json [hunk 2]
```diff
@@ -4,9 +4,9 @@
     "type": "library",
     "license": "MIT",
     "scripts": {
-        "analyse": "./vendor/bin/phpstan analyse",
-        "bench": "./vendor/bin/phpbench run --report=redis",
-        "bench:verbose": "./vendor/bin/phpbench run --report=redis --progress=blinken"
+        "analyse": "phpstan analyse",
+        "bench": "phpbench run --report=redis",
+        "bench:verbose": "phpbench run --report=redis --progress=blinken"
```
Comment (tillkruss) [inline composer.json:9]: Can you explain this change? Why use global binaries, not the vendor ones?

Comment (szepeviktor) [reply composer.json:9]: Oh noooooooo! :)

Composer is witty. Please see the purple box on https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands

File: tests/Relay.stub (hunk 1 above)
Comment (szepeviktor) [reply tests/Relay.stub:18]: I've made up all of them based on PHPStan reporting.
Their value are not `1` for sure!

File: phpstan.neon.dist [hunk 3]
```diff
@@ -0,0 +1,10 @@
+parameters:
+  # Dream level
+  #level: max
+  level: 5
+  scanFiles:
+    - tests/Relay.stub
```
[the remark this answers is not in the document]
Comment (tillkruss) [reply phpstan.neon.dist:6]: https://relay.so/stubs/relay.stub.php

File: src/Laravel/RelayConnection.php [hunk 4]
```diff
@@ -7,9 +7,6 @@
 use Illuminate\Contracts\Redis\Connection;
 use Illuminate\Redis\Connections\PhpRedisConnection;
 
-/**
- * @mixin \Relay\Relay
- */
```
[the remark this answers is not in the document]
Comment (szepeviktor) [reply src/Laravel/RelayConnection.php:12]: It is unused.

[the remark this answers is not in the document]
Comment (szepeviktor) [reply src/Laravel/RelayConnection.php:12]: No methods of `Relay\Relay` are called.

File: phpstan.neon.dist (hunk 3 above)
[the remark this answers is not in the document]
Comment (szepeviktor) [reply phpstan.neon.dist:6]: All right.
I've added `Relay` contants to it.
Could you add them too?

[the remark this answers is not in the document]
Comment (tillkruss) [reply phpstan.neon.dist]: We're working on that. As of PHP 8.1.7 class constants aren't part of the builds yet, even tho PHP-Parser 4.14.0 supports it.

**Pull request merged**